### DISCUSSION TITLE
Parallel Training of Time series Predictor

### DISCUSSION THREAD

**Discussion by vamshik113** [autogluon/autogluon#4189] [Q&A]
Hello, I've been utilizing AutoGluon and finding it quite enjoyable. I had a question about parallelizing model training. Is it possible to concurrently train all forecasters and then utilize the outputs of the best models for Ensemble Training? For instance, suppose I have an ECR Node group with multiple nodes that could be leveraged to parallelise the training of each model.

**Comment by shchur** [comment]
Hi @vamshik113, thank you for this feature request. We currently do not have distributed training for time series on our near-term roadmap, but I will convert this discussion into an issue to keep track of the progress / updates on this feature. 

If we decide to pursue this feature, we will likely need to first implement parallel local training on a single instance with Ray, and then move to distributed training next.

**Reply by vamshik113** [reply]
@shchur Thanks for your response. Alternatively, this can be implemented in Kubernetes by running each model in separate pods with model selection. Then, the weighted ensembler can be executed independently by passing data from all upstream models. To achieve this, an additional API would be required to run the weighted ensembler separately.
